app/services/schedule_processor.rb

```diff
@@ -10,16 +10,26 @@ def self.process
 
   # Step 1
   def self.process_returning_rj_retaining_their_slots
-    # Extract all returning RJs who are retaining their slots, sorted just in case
-    # Check if the schedule for the respective best(retained) slot is empty, if so save it
-    # Else update the Database that the slot isn't retained anymore
-    # But keep track of these RJs who didn't get their retained slot due to the conflict in some list
-    # We will use this conflict list for highlighting in the final CSS, so that the website users are aware of the conflict
+    @radio_jockeys = RadioJockey.all
+    @radio_jockeys.each do |jockey|
+      # Apply the condition - update only if conditions are met (This condition must be set)
+      if jockey.member_type == "Returning DJ" and jockey.retaining == "Yes"
```

Comment:
You can query this in line 13 itself using the where clause.
I also feel we should sort/order it as well, in the event some human error is made and there happens to be 2 RJs with the same retaining slot.

Reply:
This basically:
returning_rjs = RadioJockey.where(member_type: "Returning DJ", retaining: "yes").order(
      semesters_in_kanm: :desc, expected_grad: :asc, timestamp: :asc
    )